web/dataselectie/datasets/generic/views_mixins.py

```diff
@@ -396,34 +396,38 @@ def custom_aggs(self, elastic_data, request):
         """
         pass
 
-    def paginate(self, offset: int, q: dict) -> dict:
-        # Sanity check to make sure we do not pass 10000
-        if offset and settings.MAX_SEARCH_ITEMS:
-            if q['size'] + offset > settings.MAX_SEARCH_ITEMS:
-                size = settings.MAX_SEARCH_ITEMS - offset
-                q['size'] = size if size > 0 else 0
-        return q
-
     def handle_query_size_offset(self, query: dict) -> dict:
         """
-        Handles query size and offseting
+        Handles query size and offsets
         """
-        # Adding sizing if not given
+        # Adding sizing if not given. In bag we also accept page_size parameter
+        size = self.request_parameters.get('size', self.request_parameters.get('page_size', None))
+        if size:
+            try:
+                size = int(size)
+            except ValueError:
+                raise ValueError(f"Invalid size {size}")
```

Comment:
Worden deze `ValueError`s ergens opgehaald? Of krijgt gebruiker gewoon 500 terug?